runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/SimpleMonitoringInfoBuilder.java

```diff
@@ -0,0 +1,215 @@
+      spec = SimpleMonitoringInfoBuilder.specs.get(USER_COUNTER_URN_PREFIX);
+      List<String> split = Splitter.on(':').splitToList(urn);
+      if (split.size() != 4) {
+        LOG.warn(
+            "Dropping MonitoringInfo for URN %s, UserMetric namespaces and "
+                + "name cannot contain ':' characters.",
+            urn);
+        return false;
+      }
+    } else if (!SimpleMonitoringInfoBuilder.specs.containsKey(urn)) {
+      // Succeed for unknown URNs, this is an extensible metric.
+      // TODO(ajamato): Allow adding your own spec file to validate your metrics.
+      return true;
+    } else {
+      spec = SimpleMonitoringInfoBuilder.specs.get(urn);
+    }
+
+    if (!this.builder.getType().equals(spec.getTypeUrn())) {
+      LOG.warn(
+          "Dropping MonitoringInfo since for URN %s with invalid type field. Expected: %s"
+              + " Actual: %s",
+          this.builder.getUrn(), spec.getTypeUrn(), this.builder.getType());
+      return false;
+    }
+
+    Set<String> requiredLabels = new HashSet<String>(spec.getRequiredLabelsList());
+    if (!this.builder.getLabels().keySet().equals(requiredLabels)) {
+      LOG.warn(
+          "Dropping MonitoringInfo since for URN %s with invalid labels. Expected: %s"
+              + " Actual: %s",
+          this.builder.getUrn(), requiredLabels, this.builder.getLabels().keySet());
+      return false;
+    }
+    return true;
+  }
+
+  /** @return The metric URN for a user metric, with a proper URN prefix. */
+  private static String userMetricUrn(String metricNamespace, String metricName) {
+    StringBuilder sb = new StringBuilder();
+    sb.append(USER_COUNTER_URN_PREFIX);
+    sb.append(metricNamespace);
+    sb.append(':');
+    sb.append(metricName);
+    return sb.toString();
+  }
+
+  /**
+   * Sets the urn of the MonitoringInfo.
+   *
+   * @param urn The urn of the MonitoringInfo
+   */
+  public void setUrn(String urn) {
+    this.builder.setUrn(urn);
+  }
+
+  /**
+   * Sets the urn of the MonitoringInfo to a proper user metric URN for the given params.
+   *
+   * @param namespace
+   * @param name
+   */
+  public void setUrnForUserMetric(String namespace, String name) {
+    this.builder.setUrn(userMetricUrn(namespace, name));
+  }
+
+  /** Sets the timestamp of the MonitoringInfo to the current time. */
+  public void setTimestampToNow() {
+    Instant time = Instant.now();
+    this.builder.getTimestampBuilder().setSeconds(time.getEpochSecond()).setNanos(time.getNano());
+  }
+
+  /** Sets the int64Value of the CounterData in the MonitoringInfo, and the appropraite type URN. */
+  public void setInt64Value(long value) {
+    this.builder.getMetricBuilder().getCounterDataBuilder().setInt64Value(value);
+    this.builder.setType(SUM_INT64_TYPE_URN);
+  }
+
+  /**
+   * Sets the PTRANSFORM MonitoringInfo label to the given param.
+   *
+   * @param pTransform
+   */
+  public void setPTransformLabel(String pTransform) {
```

Comment:
It is better to define generalized method setLabel(String labelName, String labelValue) and provide a list of standard label names.
This is one of examples that might work
https://stackoverflow.com/questions/3978654/best-way-to-create-enum-of-strings